crates/types/src/partition_table.rs

```diff
@@ -271,18 +278,43 @@ impl PartialEq for PartitionPlacement {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize)]
+pub struct DbName(String);
+
+impl Default for DbName {
+    fn default() -> Self {
+        DbName(DB_NAME.to_owned())
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize)]
+pub struct CfName(String);
+
+impl CfName {
+    pub fn for_partition(partition_id: PartitionId) -> Self {
+        Self(format!("{PARTITION_CF_PREFIX}{partition_id}"))
+    }
+}
+
 #[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
 pub struct Partition {
+    pub log_id: LogId,
     pub key_range: RangeInclusive<PartitionKey>,
     #[serde(default)]
+    pub db_name: DbName,
+    pub cf_name: CfName,
```

Comment:
Thinking about those two strings since for the most part those will be constant everywhere, how about we make them Option (non-pub) and only set them if they are non-default values. This saves a few bytes per partition.